### ISSUE TITLE
refactor(lagrangian): replace hand-rolled BLUE update with gaussx.kalman_gain + gaussx.joseph_update

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#39]
## Problem / Request
`_blue_update` (`src/plumax/lagrangian/inversion.py:148-172`) hand-rolls the Kalman/BLUE measurement update:

```text
increment = B Fᵀ (F B Fᵀ + R)⁻¹ d
P*        = B − B Fᵀ S⁻¹ F B      (+ manual symmetrization)
```

gaussx ships both pieces: `kalman_gain(P, H, R)` (verified `K = P Hᵀ (H P Hᵀ + R)⁻¹` at `gaussx/_ssm/_kalman.py:302`) and the numerically superior Joseph-form covariance update `joseph_update(P, K, H, R)` (`(I−KH)P(I−KH)ᵀ + KRKᵀ` — guaranteed symmetric/PSD, strictly better than the difference form + re-symmetrization).

## Mathematics
The Joseph form is algebraically identical to `B − K S Kᵀ` for the exact gain but remains PSD under floating-point error, whereas the difference form can go indefinite for ill-conditioned S.

## Non-fits (checked during the audit — do not use)
`gx.blr_full_update`/`blr_diag_update` implement the Bayesian **Learning Rule** (damped natural-gradient VI), not Bayesian linear regression; `gx.conditional` is an index-partitioned joint, not an H-operator update.

## Proposed API
```python
K = gx.kalman_gain(
    lx.MatrixLinearOperator(prior_cov),
    lx.MatrixLinearOperator(jacobian),
    lx.DiagonalLinearOperator(obs_var),
)
increment = K @ innovation
posterior_cov = gx.joseph_update(prior_cov, K, jacobian, jnp.diag(obs_var))
```

## Implementation Steps
- [ ] Replace the body of `_blue_update` in `src/plumax/lagrangian/inversion.py:148-172`
- [ ] Drop the manual symmetrization
- [ ] PSD regression test on an ill-conditioned S

## Definition of Done
- [ ] Existing inversion tests pass unchanged (equivalence)
- [ ] Posterior covariance PSD under an ill-conditioned configuration
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/lagrangian/test_inversion.py::test_posterior_cov_psd_ill_conditioned`

## Relationships
- Parent (theme epic): #21
